OpenRA.Mods.Common/Activities/Resupply.cs

```diff
@@ -17,48 +17,94 @@
 
 namespace OpenRA.Mods.Common.Activities
 {
-	public class Repair : Activity
+	public class Resupply : Activity
 	{
 		readonly IHealth health;
 		readonly RepairsUnits[] allRepairsUnits;
 		readonly Target host;
 		readonly WDist closeEnough;
 		readonly Repairable repairable;
+		readonly Rearmable rearmable;
+		readonly INotifyResupply[] notifyResupplies;
 
 		int remainingTicks;
-		bool played = false;
+		bool played;
+		bool paused;
+		bool repairComplete;
+		bool rearmComplete;
+		ResupplyType activeResupplyTypes = ResupplyType.None;
 
-		public Repair(Actor self, Actor host, WDist closeEnough)
+		public Resupply(Actor self, Actor host, WDist closeEnough)
 		{
 			this.host = Target.FromActor(host);
 			this.closeEnough = closeEnough;
 			allRepairsUnits = host.TraitsImplementing<RepairsUnits>().ToArray();
 			health = self.TraitOrDefault<IHealth>();
 			repairable = self.TraitOrDefault<Repairable>();
+			rearmable = self.TraitOrDefault<Rearmable>();
```

Comment:
Do you have plans to make this more generic eventually? It would be cool if I could plug https://github.com/obrakmann/openra-fuel into this by implementing some interface.

Reply:
something like `actions = self.TraitsImplementing<IResupplyAction>().ToArray()`

Reply:
Yes, but in a follow-up.